File: framework/languages/java/zlink-framework-core/src/main/java/systems/zlink/framework/spots/ZLinkTimerTick.java
package systems.zlink.framework.spots;

import java.time.Duration;
import java.time.Instant;

public record ZLinkTimerTick(
    String name,
    long deliveryIndex,
    long scheduledIndex,
    Duration period,
    Instant scheduledAt,
    Instant startedAt,
    Duration scheduledElapsed,
    Duration startedElapsed,
    Duration delay,
    long skippedTicks) {
}

File: framework/languages/node/packages/framework/src/contracts/Timers/ZLinkTimerTick.ts
export interface ZLinkTimerTick {
  readonly name: string;
  readonly deliveryIndex: bigint;
  readonly scheduledIndex: bigint;
  readonly periodMs: number;
  readonly scheduledAt: Date;
  readonly startedAt: Date;
  readonly scheduledElapsedMs: number;
  readonly startedElapsedMs: number;
  readonly delayMs: number;
  readonly skippedTicks: bigint;
}
